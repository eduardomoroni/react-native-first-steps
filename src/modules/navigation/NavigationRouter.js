// @flow

import React from 'react'
import { Scene, Router } from 'react-native-router-flux'
import Settings from '../settings/containers/Settings'
import LoginScreen from '../login/containers/LoginScreen'
import MainScreen from '../shared/containers/MainScreen'
import Drawer, { menuDrawerProps } from './NavigationDrawer'
import CardSearchForm from '../cardSearch/containers/CardSearchForm'
import ListCards from '../cardSearch/containers/ListCards'
import CardDetails from '../cardSearch/containers/CardDetails'
import CardSwiper from '../cardSearch/containers/CardSwiper'
import WishList from '../cardTrade/containers/WishList'
import SearchResultsNavBar from '../cardSearch/containers/SearchResultsNavBar'
import I18n from 'react-native-i18n'
import { findCardsFromForm } from '../../services/CardService'
import { builtInBarStyle } from '../theme/NavBarStyle'

const testForm = {
  cardName: 'aetASDASDASD'
}

const testScreen = () => {
  const cards = findCardsFromForm(testForm)
  console.log(cards)
  return <WishList />
}

const NavigationRouter = () => {
  return (
    <Router>
      <Scene key='drawer' component={Drawer} {...menuDrawerProps} >
        <Scene key='drawerChildrenWrapper' {...builtInBarStyle} >
          <Scene key='cardSearchForm' component={CardSearchForm} title={I18n.t('card_search_form_title')} />
          <Scene key='cardDetails' component={CardDetails} />
          <Scene key='cardSwiper' component={CardSwiper} />
          <Scene key='wishList' component={WishList} />
          <Scene initial key='testScreen' component={testScreen} title='Test Screen' />
          <Scene key='mainScreen' component={MainScreen} title={I18n.t('welcome')} />
          <Scene key='settings' component={Settings} title={I18n.t('settings')} />
          <Scene key='listCards' component={ListCards} title={I18n.t('list_cards_title')} navBar={SearchResultsNavBar} />
          <Scene key='loginScreen' component={LoginScreen} title={I18n.t('login_for_title')} hideNavBar />
        </Scene>
      </Scene>
    </Router>
  )
}

export default NavigationRouter
