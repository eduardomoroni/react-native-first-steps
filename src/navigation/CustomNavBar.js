import React from 'react'
import { View, StyleSheet } from 'react-native'
import { builtInBarStyle, customBarStyle } from '../styles/NavBarStyle'

export class CustomNavBar extends React.Component {
  render () {
    const { leftRender, middleRender, rightRender } = this.props

    return (
      <View style={StyleSheet.flatten([customBarStyle.container, builtInBarStyle.navigationBarStyle])}>
        <View style={StyleSheet.flatten([customBarStyle.leftButton, customBarStyle.buttons])}>
          {leftRender}
        </View>
        <View>
          {middleRender}
        </View>
        <View style={StyleSheet.flatten([customBarStyle.rightButton, customBarStyle.buttons])}>
          {rightRender}
        </View>
      </View>
    )
  }
}

export default CustomNavBar
