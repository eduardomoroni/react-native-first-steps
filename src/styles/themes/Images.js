// leave off @2x/@3x
const images = {
  logo: require('../../assets/images/ir.png')
}

export default images
