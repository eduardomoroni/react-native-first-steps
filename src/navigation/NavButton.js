import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import { Metrics } from '../styles/themes/'
import styles from '../styles/navigation/NavBarItemsStyle'

export class NavButton extends Component {
  render () {
    const {
      iconProps,
      callback
    } = this.props

    return (
      <TouchableOpacity onPress={callback} >
        <Icon {...iconProps} />
      </TouchableOpacity>
    )
  }
}

NavButton.propTypes = {
  iconProps: PropTypes.shape({
    name: PropTypes.string,
    size: PropTypes.number,
    style: PropTypes.any
  }),
  callback: PropTypes.func
}

Icon.defaultProps = {
  name: 'heart',
  size: Metrics.icons.medium,
  style: styles.button
}
