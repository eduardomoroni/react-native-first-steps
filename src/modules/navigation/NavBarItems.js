import React from 'react'
import I18n from 'react-native-i18n'
import { Text, View } from 'react-native'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { builtInBarStyle, customBarStyle } from '../theme/NavBarStyle'
import { NavButton } from './NavButton'

type ButtonPropsType = {
  name: string,
  size: number,
  style: any
}

// TODO: Enable Flow
// TODO: REFACTOR TYPES
export const navButton = (buttonProps: ButtonPropsType, callback: Function, key: any) => {
  return (
    <NavButton iconProps={{...buttonProps}} callback={callback} key={key} />
  )
}

export const openDrawer = (component: any) => {
  NavigationActions.refresh({
    key: 'drawer',
    content: component,
    open: true,
    side: 'left'
  })
}

export const navTitle = (title: string) => {
  return (
    <Text style={builtInBarStyle.titleStyle}>
      {I18n.t(title)}
    </Text>
  )
}

// TODO: This is an array of Object ButtonPropsType
export const navButtonBar = (buttons: any) => {
  return (
    <View style={[customBarStyle.rightButton, customBarStyle.buttons]} >
      { buttons.map(i => navButton(i.buttonProps, i.callback, i.buttonProps.name)) }
    </View>
  )
}

export const navBackButton = () => {
  return navButton({name: 'angle-left'}, NavigationActions.pop)
}
