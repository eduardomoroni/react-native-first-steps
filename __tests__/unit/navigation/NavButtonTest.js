import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { NavButton } from '../../../src/modules/navigation/NavButton'
import Icon from 'react-native-vector-icons/FontAwesome'
import { Metrics } from '../../../src/modules/theme/themes/'
import styles from '../../../src/modules/theme/navigation/NavBarItemsStyle'

const props = {
  iconProps: {
    name: 'search-plus',
    size: 30,
    style: {}
  },
  callback: jest.fn()
}

describe('<NavButton />', () => {
  it('Should Render NavButton', () => {
    const wrapper = shallow(<NavButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render default Icon props', () => {
    const wrapper = shallow(<NavButton />)
    const button = wrapper.find(Icon)
    expect(button.prop('name')).toEqual('heart')
    expect(button.prop('size')).toEqual(Metrics.icons.medium)
    expect(button.prop('style')).toEqual(styles.button)
  })

  it('should fire callback on click button', () => {
    const wrapper = shallow(<NavButton {...props} />)
    wrapper.simulate('press')
    expect(props.callback).toHaveBeenCalledWith()
  })
})
