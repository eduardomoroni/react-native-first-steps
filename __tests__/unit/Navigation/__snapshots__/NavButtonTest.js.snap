// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButton /> Should Render NavButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
>
  <Icon
    name="search-plus"
    size={30}
    style={Object {}}
  />
</TouchableOpacity>
`;
